workers/proof-manager/src/worker.rs

```diff
@@ -36,8 +40,12 @@ impl Worker for ProofManager {
     where
         Self: Sized,
     {
+        ProofManager::preprocess_circuits()?;
```

Comment:
propose moving this to the start of the execution loop, perhaps before we start receiving jobs. Problem with putting this in `new` is we now explicitly block the main thread which is probably not desirable

Reply:
sure thing, good call out